lambda.js

```diff
@@ -1,16 +1,33 @@
 const server = require('restana')();
-const app = require('next')({ dev: process.env.ENV === 'dev' });
+const app = require('next')({ dev: process.env.ENV === 'dev', quiet: false });
 const files = require('serve-static');
 const path = require('path');
+const cookieParser = require('cookie-parser');
 const handle = app.getRequestHandler();
+const CheckAuth = require('./lib/use-cases/check-auth');
 
+server.use(cookieParser());
 server.use(files(path.join(__dirname, 'build')));
 server.use(files(path.join(__dirname, 'public')));
 server.use(async (req, res, next) => {
   await app.prepare();
   next();
 });
 
-server.get('*', (req, res) => handle(req, res));
+server.all('/api/*', (req, res) => handle(req, res));
 
-module.exports.handler = require('serverless-http')(server);
+server.use((req, res, next) => {
+  const checkAuth = new CheckAuth({
+    allowedGroups: process.env.allowedGroups.split(","),
+    jwt: require('jsonwebtoken')
+  });
+  if (!checkAuth.execute({ token: req.cookies.hackneyToken}) && req.url !== '/loggedout') {
+    res.writeHead(302, {'Location': '/loggedout'});
+    return res.end();
+  }
+  next();
+});
```

Comment:
We should make sure this doesn't apply to `/api` routes unless we implement some sort of CSRF protection for them. Does the ordering already solve this? I can't remember off the top of my head the order of the `server.use`s matters, so this might already be fine.

Reply:
Updated and agreed we'll fix the authorizer in the future.